src/HybridDb.Tests/Migrations/DocumentMigrationRunnerTests.cs

```diff
@@ -429,6 +429,49 @@ public async Task MigratesSelectedIdsAndIdPrefixCombined()
             migratedIds.ShouldBeLikeUnordered("aatest");
         }
 
+        [Fact]
+        public async Task DeletesRowIfMigrationReturnsNull()
+        {
+            UseTypeMapper(new AssemblyQualifiedNameTypeMapper());
+            Document<Entity>().With(x => x.Number);
+
+            var table = configuration.GetDesignFor<Entity>().Table;
+
+            void Insert(string id, int number) =>
+                store.Insert(table, id, new
+                {
+                    AwaitsReprojection = false,
+                    Discriminator = typeof(Entity).AssemblyQualifiedName,
+                    Version = 0,
+                    Document = configuration.Serializer.Serialize(new Entity()),
+                    Number = number
+                });
+
+            Insert("atest", 1);
+            Insert("aatest", 2);
+            Insert("AaAtest", 3);
+
+            ResetConfiguration();
+
+            UseTypeMapper(new AssemblyQualifiedNameTypeMapper());
+            Document<Entity>().With(x => x.Number);
+
+            UseMigrations(
+                new InlineMigration(1, new ChangeDocument<Entity>(ListOf(new IdMatcher(new[] { "aatest", "AaAtest" })),
+                    (_, _, r) => null)));

```

Comment:
I was thinking, that instead of returning null to "mark a deletion", we should return a marker object. So the migration would look something like: 

    (_, _, r) => new DeleteDocument())));
    
That way, it would take a little more effort to "accidentally" delete a document and the intention of the migration would also be clearer.